test/test_sparse.py

```diff
@@ -1746,6 +1746,16 @@ def check_device(x, device_id):
         x = self.sparse_tensor(i, v, torch.Size([3, 0]), device=1)
         check_device(x, 1)
 
+        i = self.index_tensor([[2]]).cuda(1)
+        v = self.value_tensor([5]).cuda(1)
+        x = torch.sparse_coo_tensor(i, v, torch.Size([3]))
+        check_device(x, 1)
+
+        i = self.index_tensor([[2]]).cuda(1)
+        v = self.value_empty(1, 0).cuda(1)
+        x = torch.sparse_coo_tensor(i, v, torch.Size([3, 0]))
+        check_device(x, 1)
```

Comment:
Better add a test where dtype is set here, too

Reply:
all right!